test/test_optim.py

```diff
@@ -431,6 +431,21 @@ def test_adam(self):
             with self.assertRaisesRegex(ValueError, "Invalid weight_decay value: -1"):
                 optimizer(None, lr=1e-2, weight_decay=-1)
 
+    # Test whether variance parameter is always real
+    def test_complex_adam_variance(self):
+        complex_param = torch.randn(5, 5, dtype=torch.complex64, requires_grad=True)
+        target = torch.randn(5, 5, dtype=torch.complex64)
+        optimizer = optim.Adam([complex_param], lr=0.001)
+
+        for i in range(200):
+            optimizer.zero_grad()
+            loss = torch.sum(complex_param - target).pow(2).sum()
```

Comment:
You have one extra sum here.
Also, you can use `torch.nn.functional.mse_loss()` if you want.

Reply:
oh thanks for the feedback, i am deleting the first sum. 

also, when i wanted to use `torch.nn.functional.mse_loss()` i got the error `RuntimeError: "mse_cpu" not implemented for 'ComplexFloat' `, that's why i am leaving it as `pow(2).sum()`